core/src/main/java/com/amplitude/core/utilities/FileResponseHandler.kt

```diff
@@ -9,53 +9,70 @@ import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.launch
 import org.json.JSONArray
 import org.json.JSONException
+import java.util.concurrent.atomic.AtomicBoolean
+import java.util.concurrent.atomic.AtomicInteger
 
 class FileResponseHandler(
     private val storage: EventsFileStorage,
     private val eventPipeline: EventPipeline,
     private val configuration: Configuration,
     private val scope: CoroutineScope,
     private val dispatcher: CoroutineDispatcher,
-    private val eventFilePath: String,
-    private val eventsString: String,
     private val logger: Logger?
 ) : ResponseHandler {
 
-    override fun handleSuccessResponse(successResponse: SuccessResponse) {
+    private var retries = AtomicInteger(0)
+    private var currentFlushInterval = configuration.flushIntervalMillis.toLong()
+        set(value) {
+            field = value
+            eventPipeline.flushInterval = value
+        }
+    private var backoff = AtomicBoolean(false)
+    private var currentFlushQueueSize = configuration.flushQueueSize
+        set(value) {
+            field = value
+            eventPipeline.flushQueueSize = value
+        }
+    private val maxQueueSize = 50
+
+    override fun handleSuccessResponse(successResponse: SuccessResponse, events: Any, eventsString: String) {
+        val eventFilePath = events as String
         logger?.debug("Handle response, status: ${successResponse.status}")
-        val events: List<BaseEvent>
+        val eventsList: List<BaseEvent>
         try {
-            events = JSONArray(eventsString).toEvents()
+            eventsList = JSONArray(eventsString).toEvents()
         } catch (e: JSONException) {
             storage.removeFile(eventFilePath)
             removeCallbackByInsertId(eventsString)
             throw e
         }
-        triggerEventsCallback(events, HttpStatus.SUCCESS.code, "Event sent success.")
+        triggerEventsCallback(eventsList, HttpStatus.SUCCESS.code, "Event sent success.")
         scope.launch(dispatcher) {
             storage.removeFile(eventFilePath)
         }
+        resetBackOff()
     }
 
-    override fun handleBadRequestResponse(badRequestResponse: BadRequestResponse) {
+    override fun handleBadRequestResponse(badRequestResponse: BadRequestResponse, events: Any, eventsString: String) {
```

Comment:
the input parameter `events` is a little bit confusing. 
I saw events can be String or `List<BaseEvent>`. Is it possible to specify the type instead of using Any? 
Can you please clarify when it should be a String and when it should be a List of BaseEvent objects?

Reply:
this is to cover the two type of storage, for in memory storage it would be list of events, while for file storage it would be list of file strings.

Reply:
Thanks for reminding me.